src/main/java/de/tum/in/www1/artemis/domain/Feedback.java

```diff
@@ -184,8 +185,9 @@ public void setCredits(Double credits) {
         this.credits = credits;
     }
 
+    @Nonnull
     public Boolean isPositive() {
```

Comment:
Would it be more concise if the signature is `public boolean isPositive()` instead of using the `@Nonnull` annotation?

Reply:
It would, but I now had a look in our production database: a `null` value is actually used e.g. for test cases that have not been executed or manual feedback with zero credits. So I reverted the method to return a nullable value again.

It’s very good you noticed this, as this small change would otherwise definitely have been a bug that changed how results were shown to students.